src/services/entity_repository.js

```diff
@@ -115,37 +69,61 @@ export default class EntityRepository {
       'repository.bundleStubId': null
     };
 
-    const setBundleStubIdUpdate = {
+    const updateBundleStubId = {
       $set: {
         'repository.bundleStubId': bundleStubId
       }
     };
 
-    await this.db.collection('assets').updateMany(notBundledQuery, setBundleStubIdUpdate);
-    await this.db.collection('events').updateMany(notBundledQuery, setBundleStubIdUpdate);
+    const selectedAssets = [];
+    const selectedEvents = [];
+    let usedSize = 0;
 
-    const thisBundleStubQuery = {
-      'repository.bundleStubId': bundleStubId
-    };
+    const assetsCursor = await this.db.collection('assets').find(notBundledQuery, {
+      projection: this.blacklistedFields,
+      sort: {'content.idData.timestamp': 1, assetId: 1}
+    });
+    const eventsCursor = await this.db.collection('events').find(notBundledQuery, {
+      projection: this.blacklistedFields,
+      sort: {'content.idData.timestamp': 1, eventId: 1}
+    });
 
-    const assets = await this.db
-      .collection('assets')
-      .find(thisBundleStubQuery, {projection: this.blacklistedFields})
-      .toArray();
-    const events = await this.db
-      .collection('events')
-      .find(thisBundleStubQuery, {projection: this.blacklistedFields})
-      .toArray();
+    let nextAsset = await assetsCursor.next();
+    let nextEvent = await eventsCursor.next();
 
-    const entities = this.assetsAndEventsToEntityIds(assets, events);
-    const orderedEntities = this.orderEntityIds(entities);
+    const fitsIntoSizeLimit = (entity) => {
+      const size = mongoObjectSize(entity);
+      if (usedSize + size > bundleSizeInBytesLimit) {
+        return false;
+      }
+      usedSize += size;
+      return true;
+    };
 
-    const discardedEntities = this.discardEntitiesForBundling(orderedEntities, bundleItemsCountLimit, bundleSizeInBytesLimit);
-    await this.unsetEntitiesBundlesStubs(discardedEntities);
+    while (!(nextAsset === null && nextEvent === null) && (selectedAssets.length + selectedEvents.length < bundleItemsCountLimit)) {
```

Comment:
🍝😱🍝😱🍝😱

Reply:
```suggestion
    while ((nextAsset !== null || nextEvent !== null) && (selectedAssets.length + selectedEvents.length < bundleItemsCountLimit)) {
```
Imo clearer - as it now explicitly says "Next asset or next event is not null and sum of already selected assets and events is not bigger than `bundleItemsCountLimit `"
Btw it should be `<` or `<=`?